topology.go

```diff
@@ -25,17 +26,43 @@ type TopologyOptions struct {
 	// TODO: Filter status and types in DB Query
 	Types  []string
 	Status []string
+
+	// when set to true, only the children (except the immediate children) are returned.
+	// when set to false, the direct children & the parent itself is fetched.
+	restOfTheChildrenOnly bool
```

Comment:
what do we mean by immediate, direct and rest of the children ?

The terminology is very confusing